bin/zopen-init

```diff
@@ -248,284 +234,155 @@ fi
-
-if [ -z "\$ZOPEN_QUICK_LOAD" ]; then
-  if [ -e "\$ZOPEN_ROOTFS/etc/profiled" ]; then
-    dotenvs=\$(find "\$ZOPEN_ROOTFS/etc/profiled" -type f -name 'dotenv' -print)
-    dotenvcnt=\$(echo "\$dotenvs" | wc -l | tr -d ' ')
-    filecnt=0
-    [ ! -p \$TMP_FIFO_PIPE ] || rm -f \$TMP_FIFO_PIPE
-    mkfifo \$TMP_FIFO_PIPE
-    chtag -tc 819 \$TMP_FIFO_PIPE
-    (/bin/echo "\$dotenvs" | xargs | tr ' ' '\n'>>\$TMP_FIFO_PIPE &)
-    while read FILE; do
-      [ -z "\$TMP_FIFO_PIPE" ] && break
-      filecnt=\$(expr \$filecnt + 1)
-      pct=\$(expr \$filecnt \* 100)
-      pct=\$(expr \$pct / \$dotenvcnt)
-      if [ ! "\${_BPX_TERMPATH-x}" = "OMVS" ];  then
-        /bin/echo "\047[1A\047[\$30D\047[2K- Processing \$zot configuration: \${pct}% (\${filecnt}/\${dotenvcnt})"
-      else
-        /bin/echo "Processing \$zot configuration...: \${pct}% (\${filecnt}/\${dotenvcnt})"
-      fi
-      [ -e \$FILE ] && . \$FILE
-    done < \$TMP_FIFO_PIPE
-    [ ! -z "\$TMP_FIFO_PIPE" -a ! -p \$TMP_FIFO_PIPE ] || rm -f \$TMP_FIFO_PIPE
-    if [ ! "\${_BPX_TERMPATH-x}" = "OMVS" ];  then
-      /bin/echo "\047[1A\047[\$30D\047[2K- Processed \$zot configuration: 100% (\${dotenvcnt}/\${dotenvcnt})"
-    else
-      /bin/echo "Processing \$zot configuration...: 100% (\${filecnt}/\${dotenvcnt})"
-    fi
-    unset dotenvs dotenvcnt filecnt
-  fi
-fi
-PATH=\$ZOPEN_ROOTFS/usr/local/bin:\$ZOPEN_ROOTFS/usr/bin:\$ZOPEN_ROOTFS/bin:\$ZOPEN_ROOTFS/boot:\$(sanitizeEnvVar \"\$PATH\" \":\" \"^\$ZOPEN_PKGINSTALL/.*\$\")
-export PATH=\$(deleteDuplicateEntries \"\$PATH\" \":\")
-LIBPATH=\$ZOPEN_ROOTFS/usr/local/lib:\$ZOPEN_ROOTFS/usr/lib:\$(sanitizeEnvVar "\$LIBPATH" ":" "^\$ZOPEN_PKGINSTALL/.*\$")
-export LIBPATH=\$(deleteDuplicateEntries \"\$LIBPATH\" \":\")
-MANPATH=\$ZOPEN_ROOTFS/usr/local/share/man:\$ZOPEN_ROOTFS/usr/local/share/man/\%L:\$ZOPEN_ROOTFS/usr/share/man:\$ZOPEN_ROOTFS/usr/share/man/\%L:\$(sanitizeEnvVar \"\$MANPATH\" \":\" \"^\$ZOPEN_PKGINSTALL/.*\$\")
-export MANPATH=\$(deleteDuplicateEntries \"\$MANPATH\" \":\")
+ZOPEN_CA="$ZOPEN_CA_DIR/$certFileName"
 
-unset tmp TMP_FIFO_PIPE
+printInfo "- Creating path for certificate lookups"
+[ -e "$rootfs/$ZOPEN_CA_DIR" ] || mkdir -p "$rootfs/$ZOPEN_CA_DIR"
 
-# Unset traps
-trap - EXIT INT TERM QUIT HUP
-EOF
-
-perms="744"
-printVerbose "Giving world read access to config file"
-chmod "$perms" "$configFile"
-
-effuid=$LOGNAME
-effgid=$(id -gn $effuid)
-printVerbose "Ensuring correct uid [$effuid] & gid [$effgid] for configfile if !installing as user (ie su/root/etc)"
-chown "$effuid" "$configFile" 2>/dev/null # suppress if not permitted
-chgrp "$effgid" "$configFile" 2>/dev/null # suppress if not permitted
+writeConfigFile "$configFile" "$rootfs" "$zopen_pkginstall" "$ZOPEN_CA"
 printInfo "- Created config in $configFile."
 
+export ZOPEN_ROOTFS="$rootfs"
 printInfo "- Root FS available at $rootfs"
+# Generate any system values that can be user changed
+echo "15" > "$rootfs/etc/zopen/rm_fileprocs"
```

Comment:
What does 15 represent here? Number of processes?

Reply:
It's used in the unmerge processing where the FS is scanned for obsolete links. 15 files per "thread" (or rather process); seemed to give a reasonable amount of workload distribution without spawning too many simultaneous processes for things like nurses with it's 4000+ files). That generally does need a speed up though...